python/setup.py

```diff
@@ -171,6 +171,11 @@ def initialize_options(self):
         'gandiva']
 
     def _run_cmake(self):
+        # check if build_type is correctly passed / set
+        if self.build_type not in ('release', 'debug'):
```

Comment:
I would lowercase this as sometimes it is passed as "RELEASE" or "RelWithDebInfo". Also note there are more value than this, see https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_TYPE.html

Reply:
OK on the lower case.

I can also remove the check in its entirety, if it would be too strict (don't know if those other build types are used within arrow development). I mainly wanted to address the issue described in https://issues.apache.org/jira/browse/ARROW-5165 that the error you now get is rather confusing.